server/src/main/java/org/opensearch/extensions/settings/RegisterSettingsRequest.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.extensions.settings;
+
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.io.stream.StreamOutput;
+import org.opensearch.common.settings.Setting;
+import org.opensearch.common.settings.WriteableSetting;
+import org.opensearch.transport.TransportRequest;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Request to register extension settings
```

Comment:
`settings` -> `setting` and also the class name should be `RegisterSettingRequest`

Reply:
But it's a `List<Setting<?>>` and the method that kicks it off is `getSettings()`.  I'm okay with adding a word like "custom" here, but it's most definitely plural.

Reply:
This makes sense. Thinking out loud: we need a way to differentiate `setting` vs `settings` as `settings` are already available for reading/writing from a stream.

Reply:
I agree, we're really overloading the term!  Ultimately they are similar to Properties (with more functionality) so I'm fine with Settings as the class name but we do need to use another term for variable naming and javadocs.

Reply:
Done.